DllImportGenerator/DllImportGenerator/Marshalling/StringMarshaller.Utf8.cs

```diff
@@ -0,0 +1,280 @@
+                                                InvocationExpression(
+                                                    MemberAccessExpression(
+                                                        SyntaxKind.SimpleMemberAccessExpression,
+                                                        UTF8EncodingType,
+                                                        IdentifierName("GetBytes")),
+                                                    ArgumentList(
+                                                        SeparatedList(new ArgumentSyntax[] {
+                                                            Argument(IdentifierName(fixedCharIdentifier)),
+                                                            Argument(MemberAccessExpression(
+                                                                SyntaxKind.SimpleMemberAccessExpression,
+                                                                IdentifierName(managedIdentifier),
+                                                                IdentifierName("Length"))),
+                                                            Argument(IdentifierName(stackAllocPtrIdentifier)),
+                                                            Argument(IdentifierName(byteLenIdentifier))}))))))),
+                                // <stackAllocPtr>[<byteLen>] = 0;
+                                ExpressionStatement(
+                                    AssignmentExpression(
+                                        SyntaxKind.SimpleAssignmentExpression,
+                                        ElementAccessExpression(
+                                            IdentifierName(stackAllocPtrIdentifier),
+                                            BracketedArgumentList(
+                                                SingletonSeparatedList<ArgumentSyntax>(
+                                                    Argument(IdentifierName(byteLenIdentifier))))),
+                                        LiteralExpression(SyntaxKind.NumericLiteralExpression, Literal(0)))),
+                                // <nativeIdentifier> = <stackAllocPtr>;
+                                ExpressionStatement(
+                                    AssignmentExpression(
+                                        SyntaxKind.SimpleAssignmentExpression,
+                                        IdentifierName(nativeIdentifier),
+                                        CastExpression(
+                                            NativeType,
+                                            IdentifierName(stackAllocPtrIdentifier)))));
+
+                            // if (<managed> == null)
+                            // {
+                            //     <native> = null;
+                            // }
+                            // else
+                            // {
+                            //     int <byteLen> = (<managed>.Length + 1) * 3 + 1;
```

Comment:
Maybe use `Encoding.UTF8.GetMaxByteCount` or a comment explaining this calculation?

Reply:
Huh, I thought I had a comment. Must have lost it during some moving around of things...
I had kept it as the straight calculation to match what built-in and avoid the extra call and checks that `GetMaxByteCount` does.

Reply:
If that is the plan, then let's call `GetMaxByteCount(1)` during the source gen and use that as the constant.

Reply:
Bah, so the logic for the calculation is:
- `+ 1` for number of characters in case left over high surrogate is ?
- `* <MaxByteCountPerChar>`
- `+ 1` for null terminator

`GetMaxByteCount` takes into account the +1 for the number of characters, so `GetMaxByteCount(1)` would not give us the constant we want to multiply by.